support/mender-inventory-provides

```diff
@@ -10,6 +10,25 @@ set -e
 # Also submit device_type, which is in a separate file, not in the database. We don't know if the
 # file contains a newline, so it's important that this is done last, to avoid corrupting other
 # entries.
-grep '^device_type=' /var/lib/mender/device_type
+
+# Set default value for DeviceTypeFile
+default_device_type_file="/var/lib/mender/device_type"
+
+# Read mender.conf file to find the DeviceTypeFile key if it exists prioritizing /etc over /var/lib
+if grep -q '"DeviceTypeFile"' /etc/mender/mender.conf; then
+    device_type_file=$(grep '"DeviceTypeFile"' /etc/mender/mender.conf | awk -F ': ' '{print $2}' | tr -d '"' | tr -d ',')
```

Comment:
This command is not robust if, for example, you have a (valid!) json string in one line like `"DeviceTypeFile": "/my/file", "SomethingElse": "OtherValue"`. See how we do this in [`rootfs-image` update module](https://github.com/mendersoftware/mender/blob/4.0.5/support/modules/rootfs-image#L62-L68) when no `jq` is present. Here we could have something like:

```
MATCH="[Dd][Ee][Vv][Ii][Cc][Ee][Tt][Yy][Pp][Ee][Ff][Ii][Ll][Ee]"
device_type_file="$(sed -ne '/"'"$MATCH"'" *: *"[^"]*"/ { s/.*"'"$MATCH"'" *: *"\([^"]*\)".*/\1/; p }' "/etc/mender/mender.conf" || true)"
```

Reply:
It does make sense. Thanks.